whisper/stt/processing/vad.py

```diff
@@ -0,0 +1,390 @@
+
+        segments = get_speech_timestamps(audio, vad_options=options)
+    else:
+        segments = get_vad_segments(
+            audio,
+            sample_rate=sample_rate,
+            output_sample=True,
+            min_speech_duration=min_speech_duration,
+            min_silence_duration=min_silence_duration,
+            dilatation=dilatation,
+            method=method,
+        )
+    segments = [(seg["start"], seg["end"]) for seg in segments]
+    if len(segments) == 0:
+        if avoid_empty_speech:
+            segments = [(0, audio.shape[-1])]
+        else:
+            np.array([]), [], lambda t, t2=None: t if t2 is None else [t, t2]
+
+    audio_speech = np.concatenate([audio[..., s:e] for s, e in segments], axis=-1)
+    # audio_speech = torch.cat([audio[..., s:e] for s,e in segments], dim=-1)
+
+    if not use_sample:
+        segments = [
+            (float(s) / sample_rate, float(e) / sample_rate) for s, e in segments
+        ]
+    if return_format == "dict":
+        segments = [{"start": s, "end": e} for s, e in segments]
+    return (
+        audio_speech,
+        segments,
+        lambda t, t2=None: do_convert_timestamps(segments, t, t2),
+    )
+
+
+def do_convert_timestamps(segments, t, t2=None):
+    """
+    Convert timestamp from audio without non-speech segments to original audio (with non-speech segments)
+
+    parameters:
+        segments: list of tuple (start, end) corresponding to non-speech segments in original audio
+        t: timestamp to convert
+        t2: second timestamp to convert (optional), when the two timestamps should be in the same segment
+    """
+    assert len(segments)
+    ioffset = 0  # Input offset
+    ooffset = 0  # Output offset
+    ipreviousend = 0
+    result = []
+    for istart, iend in segments:
+        ostart = ooffset
+        oend = ostart + (iend - istart)
+        ooffset = oend
+        ioffset += istart - ipreviousend
+        ipreviousend = iend
+        t_in = t <= oend
+        t2_in = t_in if t2 is None else t2 <= oend
+        if t_in or t2_in:
+            result.append(
+                [
+                    max(istart, min(iend, ioffset + t)),
+                    max(istart, min(iend, ioffset + t2)) if t2 is not None else None,
+                ]
+            )
+            if t_in and t2_in:
+                break
+    if not len(result):
+        result.append([ioffset + t, ioffset + t2 if t2 is not None else None])
+
+    if len(result) > 1:
+        # Minimize difference between durations
+        result = sorted(result, key=lambda x: abs(abs(t2 - t) - abs(x[1] - x[0])))
+    result = result[0]
+    if t2 is None:
+        result = round(result[0], 2)
+    else:
+        result = [round(x, 2) for x in result]
+    return result
+
+
+def get_vad_segments(
+    audio,
+    sample_rate=16000,
+    output_sample=False,
+    min_speech_duration=0.1,
+    min_silence_duration=0.1,
+    dilatation=0.5,
```

Comment:
Those could be parameterizable as environment variables as well
(VAD_DILATATION, ...)